src/services/validator.ts

```diff
@@ -0,0 +1,75 @@
+function isIsraeliIdValid(id: string) {
+  if (typeof id !== "string") return false;
+  const trimmed = id.trim();
+
+  if (!trimmed) return false;
+  if (!/^\d{9}$/.test(trimmed)) return false;
+
+  let counter = 0;
+  for (let i = 0; i < 9; i++) {
+    let num = parseInt(trimmed[i]) * ((i % 2) + 1);
+    if (num > 9) num -= 9;
+    counter += num;
+  }
+  return counter % 10 === 0;
+}
+
+function isValidPhone(phone: string) {
+  const pattern = /^\+9725\d{8}$/;
+  return pattern.test(phone);
+}
+
+export const validator = {
```

Comment:
i suggest youll think about more generic implementation of validaor service for each field

Reply:
I added a pre-sign up trigger that handle the validation more nicely